.werft/installer-tests.ts

```diff
@@ -238,60 +207,120 @@ installerTests(TEST_CONFIGURATIONS[testConfig]).catch((err) => {
 
 export async function installerTests(config: TestConfig) {
     console.log(config.DESCRIPTION);
+    // these phases set up the infrastructure
+    werft.phase(`create-${cloud}-infra`, `Create the infrastructure in ${cloud}`);
     for (let phase of config.PHASES) {
         const phaseSteps = INFRA_PHASES[phase];
         const ret = callMakeTargets(phaseSteps.phase, phaseSteps.description, phaseSteps.makeTarget);
         if (ret) {
-            // there is not point in continuing if one stage fails
-            // TODO: maybe add failable, phases
+            // there is not point in continuing if one stage fails for infra setup
+            werft.fail(`create-${cloud}-infra`, "Cluster creation failed");
             break;
         }
     }
+    werft.done(`create-${cloud}-infra`);
+
+    if (upgrade && upgrade == "true") {
```

Comment:
Isn't this just the same as `if (upgrade === "true")`

```suggestion
if (upgrade === "true") {
```